modules/consensus/accept.go

```diff
@@ -207,78 +195,110 @@ func (cs *ConsensusSet) addBlockToTree(b types.Block) (ce changeEntry, err error
 // This method is typically only be used when there would otherwise be multiple
 // consecutive calls to AcceptBlock with each successive call accepting the
 // child block of the previous call.
-func (cs *ConsensusSet) managedAcceptBlock(b types.Block) error {
-	// Grab a lock on the consensus set. Lock is demoted later in the function,
-	// failure to unlock before returning an error will cause a deadlock.
+func (cs *ConsensusSet) managedAcceptBlocks(blocks []types.Block) (bool, error) {
+	// Grab a lock on the consensus set.
 	cs.mu.Lock()
+	defer cs.mu.Unlock()
+
+	// Make sure that blocks are consecutive.
+	for i, b := range blocks {
```

Comment:
For the time being, I think that this is a good requirement to have. It's not strictly necessary, and will probably be changed once we are doing headers-first downloading and parallel block downloading, but if we get rid of it now there will be more code complexity needed to maintain correctness.